spec/lib/saml_idp/request_spec.rb

```diff
@@ -479,6 +479,146 @@ module SamlIdp
       end
     end
 
+    describe '#cert_errors' do
+      let(:saml_request) { make_saml_request }
+
+      subject do
+        described_class.from_deflated_request saml_request
+      end
+
+      describe 'document is not signed' do
+        it 'returns nil' do
+          expect(subject.cert_errors).to be_nil
+        end
+      end
+
+      describe 'document is signed' do
+        let(:saml_request) { signed_auth_request }
+        let(:service_provider) { subject.service_provider }
+        let(:cert) { saml_settings.get_sp_cert }
+
+        describe 'the service provider has no registered certs' do
+          before { subject.service_provider.certs = [] }
+
```

Comment:
rubocop might yell at this extra space?
```suggestion
          before { subject.service_provider.certs = [] }
```

Reply:
it didn't! but thanks for pointing it out, i removed the extra newline in https://github.com/18F/saml_idp/pull/108/commits/4d6da48ecaabbe8be0d06577d37b3c559902ad6f